File: test_constructor.py
import sys
import os
root_path = os.path.dirname(os.path.abspath(__file__))
build_path = os.path.join(root_path, 'build')
sys.path.insert(0, build_path)

import preprocessor

planet = preprocessor.world(5.0, 6.0)
planet2 = preprocessor.world("hello world")

print planet.sum_s()
print planet2.greet()

File: src/test_constructor.cpp
#include<string>
#include<boost/python.hpp>

using namespace std;
using namespace boost::python;

struct World{
    World(string msg):msg(msg){}
    World(double a, double b):a(a), b(b){}
    void set(string msg) {this->msg = msg;}
    string greet() {return msg;}
    double sum_s() {return a+b;}
    string msg;
    double a;
    double b;
};

BOOST_PYTHON_MODULE(preprocessor)
{
    class_<World>("world", init<string>())
        .def(init<double, double>())
        .def("greet", &World::greet)
        .def("set", &World::set)
        .def("sum_s", &World::sum_s);
}